src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/IncrementalGeneratorInitializationContextExtensions.cs

```diff
@@ -27,26 +28,29 @@ public static void RegisterDiagnostics(this IncrementalGeneratorInitializationCo
         public static void RegisterConcatenatedSyntaxOutputs<TNode>(this IncrementalGeneratorInitializationContext context, IncrementalValuesProvider<TNode> nodes, string fileName)
             where TNode : SyntaxNode
         {
-            IncrementalValueProvider<string> generatedMethods = nodes
+            IncrementalValueProvider<ImmutableArray<string>> generatedMethods = nodes
                 .Select(
                     static (node, ct) => node.NormalizeWhitespace().ToFullString())
-                .Collect()
-                .Select(static (generatedSources, ct) =>
+                .Collect();
+
+            context.RegisterSourceOutput(generatedMethods,
+                (context, generatedSources) =>
                 {
+                    // Don't generate a file if we don't have to, to avoid the extra IDE overhead once we have generated
+                    // files in play.
+                    if (generatedSources.IsEmpty)
+                        return;
```

Comment:
Nit: this method is now mixing block brace styles... the foreach below uses braces, this if doesn't.  It'd be nice to add braces here, too.